From fbe6de4d162644aa07b14e93aa2f01299bf4692f Mon Sep 17 00:00:00 2001
From: Jan Friesse <jfriesse@redhat.com>
Date: Wed, 1 Feb 2012 17:45:35 +0100
Subject: [PATCH] Wait for corosync-notifyd exit in init script

Without wait for real exit of corosync-notifyd it can happen, that new
corosync-notifyd is killed. To prevent such condition, stop now wait for
process to die, before exit of stop function.

Signed-off-by: Jan Friesse <jfriesse@redhat.com>
Reviewed-by: Steven Dake <sdake@redhat.com>
(cherry picked from commit 8e584591760704a0fe0398e39a10646f2ddaa2ba)
---
 init/notifyd.in |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/init/notifyd.in b/init/notifyd.in
index 86c6393..84de9ae 100755
--- a/init/notifyd.in
+++ b/init/notifyd.in
@@ -106,6 +106,12 @@ stop()
 	success
 	echo
 
+	echo -n "Waiting for $prog services to unload:"
+	while status $prog > /dev/null 2>&1; do
+		sleep 1
+		echo -n "."
+	done
+
 	rm -f $LOCK_FILE
 	success
 	echo
-- 
1.7.1

